lib/publiccloud/instance.pm

```diff
@@ -349,56 +351,59 @@ sub wait_for_ssh {
     $args{ignore_wrong_pubkey} //= 0;
     $args{username} //= $self->username();
     $args{public_ip} //= $self->public_ip();
+    $args{systemup_check} //= 1;    # check also system up, defult:true; when false,only ssh checked.
     my $start_time = time();
-    my $check_port = 1;
-    my $sleep_period = $args{ignore_wrong_pubkey} ? 20 : 1;
-
-    # Looping until reaching timeout or passing two conditions :
-    # - SSH port 22 is reachable
-    # - journalctl got message about reaching one of certain targets
-    while ((my $duration = time() - $start_time) < $args{timeout}) {
-        if ($check_port) {
-            $check_port = 0 if (script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0);
+    my $instance_msg = sprintf("instance %s with public IP:%s within %d seconds", $self->{instance_id}, $self->{public_ip}, $args{timeout});
+    my $status = 0;
+    # Looping until SSH port 22 is reachable or timeout:
+    script_retry('nc -vz -w 1 ' . $self->{public_ip} . ' 22',
+        timeout => $args{timeout}, delay => ($args{ignore_wrong_pubkey} ? 20 : 1), retry => 1, fail_msg => "Unable to reach SSH port of " . $instance_msg);
```

Comment:
I'd use either `"Unable to reach SSH port of $instance_msg"` or `'Unable to reach SSH port of ' . $instance_msg'`

Reply:
done